pkg/common/handler.go

```diff
@@ -95,13 +95,17 @@ func (h *Handler) handleEvent(ctx context.Context, payload interface{}, hdr http
 	h.Logger.Infof("Handling %s", gitHubEventType)
 
 	cloudEventType := sourcesv1alpha1.GitHubEventType(gitHubEventType)
-	subject := SubjectFromGitHubEvent(gh.Event(gitHubEventType), payload, h.Logger)
+	subject, extensions := SubjectAndExtensionsFromGitHubEvent(gh.Event(gitHubEventType), payload, h.Logger)
```

Comment:
I would have made a new method for the extensions and passed the event to it as a EventWriter or some small interface with the SetExtension method expected.